test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/net/URLDecoderTest.scala

```diff
@@ -0,0 +1,74 @@
+package org.scalajs.testsuite.javalib.math
+
+import java.net.URLDecoder
+import java.io.UnsupportedEncodingException
+import org.scalajs.jasminetest.JasmineTest
+
+object URLDecoderTest extends JasmineTest {
```

Comment:
New tests should be written in JUnit in `test-suite/shared/`, if they're supposed to behave in the same way on JS and JVM.